spec/datadog/core/telemetry/event_spec.rb

```diff
@@ -62,6 +65,26 @@
         end
       end
 
+      ['generate-metrics', 'distributions'].each do |request_type|
+        context request_type do
+          let(:request_type) { request_type }
+
+          context 'when payload is nil' do
+            let(:payload) { nil }
+
+            it 'raise ArgumentError' do
+              expect { telemetry_request }.to raise_error(ArgumentError)
```

Comment:
Minor: When reusing system exceptions, it's not a bad idea to try to match a bit on the message, so that we don't accidentally make the test pass by triggering another `ArgumentError` in the test code itself (e.g. if we pass the wrong number of arguments to the test code, for instance)

```suggestion
            it 'raises an ArgumentError' do
              expect { telemetry_request }.to raise_error(ArgumentError, /needs to provide the payload/)
```